src/tools/changeReportStatus.js

```diff
@@ -13,21 +14,25 @@ export default async function changeReportStatus(ids, status) {
   const idsArray = ids.split(',');
 
   const promises = [];
+  await sequelize.transaction(async (transaction) => {
+    for await (const id of idsArray) {
+      const report = await ActivityReport.unscoped().findOne({ where: { id } });
 
-  for await (const id of idsArray) {
-    const report = await ActivityReport.unscoped().findOne({ where: { id } });
-
-    if (report) {
-      auditLogger.info(`Changing status of report: ${id} to ${status}`);
-      promises.push(
-        report.update({
-          submissionStatus: status,
-        }),
-      );
-    } else {
-      auditLogger.info(`Couldn't find any reports with the id: ${id}`);
+      if (report) {
+        auditLogger.info(`Changing status of report: ${id} to ${status}`);
+        promises.push(
+          await report.update(
```

Comment:
I think we either need to remove the `await` from the `report.update` call or remove the `promises` array altogether. Calling `await report.update(...)` will wait for the promise to return before moving on to the next id in `idsArray`. The `promise.all` at the end does nothing because all promises in the array have already resolved (because they were await'ed).